packaging/linux/post_install.sh

```diff
@@ -37,6 +33,68 @@ make_mountpoint() {
   fi
 }
 
+# We use machinectl here because systemctl commands do not work in either a su shell or with systemd-run --uid.
+exec_as() {
+    user=$1
+    shift
+    machinectl -q shell "$user"@ "$@"
+}
+
+# Exits with 0 iff the given user is running the service with systemd
+systemd_unit_active_for() {
+    user=$1
+    service=$2
+
+    # exit with 1 if command fails (for example, if not using systemd)
+    active_status="$(exec_as "$user" /usr/bin/systemctl --user is-active "$service")" || return "$?"
+
+    # exit code lost by machinectl, so grep stdout
+    # use -w so "active" needle doesn't match "inactive"
+    echo "$active_status" | grep -wq "active"
+}
+
+systemctl_restart_as() {
+    user=$1
+    service=$2
+    echo "Restarting $service for $user..."
+    exec_as "$user" /usr/bin/systemctl --user restart "$service"
+}
+
+systemd_safe_restart_services() {
+    pids="$(pidof /usr/bin/keybase)"
+    while read -r pid; do
+        user="$(ps -o user= -p "$pid")"
+
+        if systemd_unit_active_for "$user" "kbfs.service"; then
+            # KBFS running, need to check mounts before proceeding
+
+            # Don't use `mount(8)` to find kbfs mounts in case there are non-kbfs fuse systems
+            mount="$(exec_as "$user" /usr/bin/keybase config get -b mountdir | tr -d '\r')"
+            uses="$(exec_as "$user" /usr/bin/lsof "$mount")"
+            if [ -n "$uses" ]; then
+                using_programs="$(echo "$uses" | tail -n +2 | awk '{print $1}' | tr '\n' ', ')"
+                echo "KBFS mount $mount for user $user currently in use by ($using_programs)."
+                echo "$uses"
```

Comment:
Doesn't the previous line tell us what we need to know?

Reply:
Yeah they both work, The first line just says `tail, awk` etc. and this one says the entire lsof output. I can get rid of one or both if you like, or maybe add some more explanation.

Reply:
I guess I'd have to see sample output, but seems like you can cut this line and not lose anything important, if there wasn't an error.